x-pack/solutions/observability/plugins/apm/public/components/app/settings/custom_link/create_edit_custom_link_flyout/filters_section.tsx

```diff
@@ -16,12 +16,12 @@ import {
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { isEmpty } from 'lodash';
-import React from 'react';
+import React, { useMemo } from 'react';
 import type { Filter, FilterKey } from '../../../../../../common/custom_link/custom_link_types';
 import { DEFAULT_OPTION, FILTER_SELECT_OPTIONS, getSelectOptions } from './helper';
 import { SuggestionsSelect } from '../../../../shared/suggestions_select';
 
-export function FiltersSection({
+export const FiltersSection = React.memo(function FiltersSection({
```

Comment:
I don't think we need to `React.memo` this component, do we?

I checked the profiler, and it seems like the cause for the `SuggestionsSelect` re-render is the `start`, `end`, and `onChange` props.

You've already memoized the `start` and `end` props, and that apparently already ensures the component doesn't trigger the `useFetcher`. You could wrap the `onChangeFilter` in a `useCallback` too, and this will ensure the component doesn't re-render.